.github/workflows/run_periodic_tests.yml

```diff
@@ -38,42 +38,58 @@ jobs:
           python -m pip install pre-commit
           pre-commit run -a
 
-  build:
+  run_unit_tests:
     needs: style
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-12, macos-14, windows-latest]
         python-version: ["3.9", "3.10", "3.11", "3.12"]
+        exclude:
+          - os: ubuntu-latest
+            python-version: "3.12"
     steps:
       - name: Check out PyBaMM repository
         uses: actions/checkout@v4
-        with:
-          fetch-depth: 0
-
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v5
-        with:
-          python-version: ${{ matrix.python-version }}
 
       - name: Install Linux system dependencies
         if: matrix.os == 'ubuntu-latest'
         run: |
           sudo apt-get update
-          sudo apt install gfortran gcc libopenblas-dev graphviz pandoc
-          sudo apt install texlive-full
+          sudo apt install gfortran gcc graphviz pandoc
+
+      # dot -c is for registering graphviz fonts and plugins
+      - name: Install OpenBLAS and TeXLive for Linux
+        if: matrix.os == 'ubuntu-latest'
+        run: |
+          sudo apt-get update
+          sudo dot -c
+          sudo apt-get install libopenblas-dev texlive-latex-extra dvipng
```

Comment:
```suggestion
```

We were installing OpenBLAS and TeXLive separately because they had problems with getting cached. We can keep them in the same step for the scheduled jobs.